src/components/publicMap.js

```diff
@@ -27,23 +35,36 @@ const MapboxGLMap = (props) => {
       mapboxgl: mapboxgl
     }));
     setMap(map)
+    
```

Comment:
you need to remove the map when the component unmounts
```suggestion
return () => map.remove();
```